scripts/west_commands/runners/nrfjprog.py

```diff
@@ -113,8 +114,46 @@ def do_run(self, command, **kwargs):
                              format(self.hex_) +
                              'Try enabling CONFIG_BUILD_OUTPUT_HEX.')
 
+        if self.family == 'NRF53':
+            # self.hex_ can contain code for both application core and network
+            # core. Code being flashed to the network core needs to have a
+            # specific argument provided to nrfjprog. If network code is found,
+            # generate two new hex files, one for each core, and flash them
+            # with the correct '--coprocessor' argument.
+            ih = IntelHex(self.hex_)
+            net_hex = IntelHex()
+            app_hex = IntelHex()
+            for s in ih.segments():
+                if s[0] >= 0x01000000:
+                    net_hex.merge(ih[s[0]:s[1]])
+                else:
+                    app_hex.merge(ih[s[0]:s[1]])
+
+            if len(net_hex) > 0:
+                wd = os.path.dirname(self.hex_)
+                net_hex_file = os.path.join(wd, 'GENERATED_CP_NETWORK_'
```

Comment:
This hex file might not be in the build directory, so I'm not sure about creating it in `wd`. Perhaps you can use the `tempfile` module instead?

Doing that will probably complicate the testing a little bit, since you won't be able to use RC_KERNEL_HEX as easily. I'm glad to discuss further if you want.